sklearn_numba_dpex/kmeans/engine.py

```diff
@@ -61,6 +61,12 @@ class KMeansEngine(KMeansCythonEngine):
 
     """
 
+    # NB: by convention, all methods that are not prefixed with an underscore `_` in
+    # this class (let's call it *public methods*) override the corresponding method
+    # from sklearn.cluster._kmeans.KMeansCythonEngine. All methods that do not override
+    # a method in sklearn.cluster._kmeans.KMeansCythonEngine are considered private to
+    # this implementation and prefixed with `_`.
```

Comment:
Thank you for adding this comment!

As private and public methods are conventionally defined as such, I think this can be boiled down to:

```suggestion
    # NB: by convention, all public methods override the corresponding method
    # from `sklearn.cluster._kmeans.KMeansCythonEngine`.
    # All methods that do not override a method in
    # `sklearn.cluster._kmeans.KMeansCythonEngine` are considered private to
    # this implementation.
```

What do you think?